File: functions/update_budget.c
#include <stdio.h>
float income, rent, utilities, grocereis, transportation, savings, expenses, spend;

float input(char type[], float var){
    printf("Monthly %s:\n",type);
    scanf("%f", &var);
return var; 
}

void percent(char type[], int amount){
    int per = amount/income*100;

    printf("Your %s is %d%% of your income.\n", type, per);
}


int main(void){
    printf("This is a budget calculator,\n");
    income = input("income", income);
    rent = input("rent", rent);
    utilities = input("utilities", utilities);
    grocereis = input("grocereis", grocereis);
    transportation = input("transportation", transportation);
    savings = income * .2;
    expenses = rent+ utilities + grocereis + transportation;
    spend = income - expenses - savings;
    printf("You make %.2f\n", income);
    printf("You savings are  $%.2f\n", savings);
    printf("You expenses are $%.2f\n", expenses); 
    printf("You spending money is $%.2f\n", spend);

    percent("rent", rent);
    percent("utilities", utilities);
    percent("grocereis", grocereis);
    percent("transporataion", transportation);
    percent("expenses", expenses);
    percent("savings", savings);
    percent("spend", spend);

return 0;
}

File: functions/update_budget.py
print("Hello, and welcome to your financial calculator!\n")
income=float(input("What is your monthly income:"))
rent=float(input("What is your monthly rent: "))
utilities=float(input("What is your monthly utilities: "))
groceries=float(input("What is your monthly groceries: "))
transportation=float(input("What is your monthly transportation costs: "))
saving = income*2
expenses = rent+ utilities + groceries + transportation
spend = income - expenses - saving

def percent(type, amount):
    per = amount/income *100

    return(f"Your {type} is  {per}% income")

print(f"Your monthly income is ${income:.2f}\n")
print(f"Your monthly expenses are ${expenses:.2f}\n")
print(f"Your monthly savings is ${saving:.2f}\n")
print(f"Your monthly spending money is ${spend:.2f}\n")
print(percent("rent",rent))
print(percent("utilities", utilities))
print(percent("groceries", groceries))
print(percent("transportation", transportation))
print(percent("saving", saving))
print(percent("expenses", expenses))